src/sentry/silo/client.py

```diff
@@ -121,3 +181,10 @@ def __init__(self, region: Region) -> None:
         # Ensure the region is registered
         self.region = get_region_by_name(region.name)
         self.base_url = self.region.address
+
+    def build_session(self) -> SafeSession:
+        """
+        Generates a safe Requests session for the API client to use.
+        This injects a custom is_ipaddress_permitted function to allow only connections to Region Silo IP addresses.
+        """
+        return build_session(is_ipaddress_permitted=validate_region_ip_address)
```

Comment:
I this is an improvement over what we have now as we can now allow region clients to send requests to control. I almost want a way to jail RPC calls and forwarded hooks to only be able to go to known allowed hosts.